src/external.jl

```diff
@@ -0,0 +1,152 @@
+function fix_external_callback end
+
+mutable struct FixExternal
+    const lmp::LMP
+    const name::String
+    const callback
+    timestep::Int
+    nlocal::Int
+    ids::Union{Nothing, Vector{Cint}}
+    x::Union{Nothing, Matrix{Float64}}
+    f::Union{Nothing, Matrix{Float64}}
```

Comment:
Why store the fields here? As far as I can tell the state is only used during the execution of the callback?